### PR TITLE
System Broker Component

### PR DESCRIPTION
**Pull request by danielgospodinow**
**Description**

Changes proposed in this pull request:

- Introduce System Broker component

### CODE REVIEW

File: components/system-broker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM golang:1.15.5-alpine3.12 as builder
```
Comment (desislavaa) [inline components/system-broker/Dockerfile:1]: Increase Go version

File: components/system-broker/Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+APP_NAME = compass-system-broker
+APP_PATH = components/system-broker
+BUILDPACK = eu.gcr.io/kyma-project/test-infra/buildpack-golang-toolbox:v20201208-f3aa8f74
```
Comment (desislavaa) [inline components/system-broker/Makefile:3]: Increase Go version

**Review by KirilKabakchiev — COMMENTED** [review]
Don't forget to include regexes for the two new queries in the packagetobundle middleware

File: components/system-broker/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+Thus, an example call to local broker would be:
+
+`curl -H "X-Broker-API-Version: 2.15" -H "Authorization: Bearer eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJzY29wZXMiOiJhcHBsaWNhdGlvbjpyZWFkIGF1dG9tYXRpY19zY2VuYXJpb19hc3NpZ25tZW50OndyaXRlIGF1dG9tYXRpY19zY2VuYXJpb19hc3NpZ25tZW50OnJlYWQgaGVhbHRoX2NoZWNrczpyZWFkIGFwcGxpY2F0aW9uOndyaXRlIHJ1bnRpbWU6d3JpdGUgbGFiZWxfZGVmaW5pdGlvbjp3cml0ZSBsYWJlbF9kZWZpbml0aW9uOnJlYWQgcnVudGltZTpyZWFkIHRlbmFudDpyZWFkIiwidGVuYW50IjoiM2U2NGViYWUtMzhiNS00NmEwLWIxZWQtOWNjZWUxNTNhMGFlIn0." localhost:8080/broker/v2/catalog`
+
+In order to run full Kyma on minikube and Compass with system broker:
+
+1. Checkout `system-broker`, navigate to `compass/installation/cmd` and execute `./run.sh --kyma-installation full`. Specify `--kyma-installation minimal` if you want to run just Compass with minimal Kyma dependencies.
+2. When the installer prints out `Status: InProgress, description: install component compass` navigate to the `system-broker` directory and execute `make deploy-on-minikube`
+3. Checkout `global_applications` branch, navigate to the `director` directory and execute `make deploy-on-minikube`
```
Comment (KirilKabakchiev) [inline components/system-broker/README.md:25]: this is outdated, please update to reflect current requirements to run locally

File: components/system-broker/README.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+The System Broker binary allows to override some configuration parameters. You can specify following as command line flags or as environment variables.
+
+[Config struct](https://github.com/kyma-incubator/compass/blob/0f0eeb38e7a5d8db655b6870138e5add257ebb1d/components/system-broker/internal/config/config.go#L30) is self descriptive. Default values can be found in `DefaultConfig` method. 
+Example for using env variables can be found in the [helm charts](https://github.com/kyma-incubator/compass/blob/4b49dae2cce65f0efa98d0a9e664ae65c0f059f8/chart/compass/charts/system-broker/templates/deployment.yaml#L53).
+
+## Usage
+
+Provision and deprovision are async and require `accepts_incomplete=true`. Bind and unbind are sync.
+
+### Specifications
```
Comment (KirilKabakchiev) [inline components/system-broker/README.md:40]: this api was moved to ORD service

File: components/system-broker/internal/specs/api.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+type SpecsHandler struct {
+	specsFetcher SpecsFetcher
+	rootAPI      string
+}
+
+func (h *SpecsHandler) Routes(router *mux.Router) {
+	router.HandleFunc(h.rootAPI+SpecsAPI, h.FetchSpec).Methods(http.MethodGet)
+}
+
+func (h *SpecsHandler) FetchSpec(w http.ResponseWriter, req *http.Request) {
```
Comment (KirilKabakchiev) [inline components/system-broker/internal/specs/api.go:44]: check whether this api behaves the same as the specs api in ORD service and if so, we should probably only have the one there. Referencing it in here (in the OSB metadata of the catalog would require, injecting relevant env vars pointing to ORD svc as well as aligning the security access, which might require some alignment, discussions)

Comment (danielgospodinow) [reply components/system-broker/internal/specs/api.go:44]: Absolutely, maybe it's a good idea to remove the System Broker Specs API in favour of ORD's Specs API but I think that a fair amount of work should be done in regards to this task, so maybe I think that it should be a separate one. Maybe it's out-of-scope for this productisation stage. What do you think?

Comment (KirilKabakchiev) [reply components/system-broker/internal/specs/api.go:44]: Personally, either is fine by me. @dpanayotov ?

Comment (dpanayotov) [reply components/system-broker/internal/specs/api.go:44]: out of scope here, will be addressed in another BLI

**Review by KirilKabakchiev — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]
readme file looks good

**Review by KirilKabakchiev — APPROVED** [review]

**Pull request merged**